tlatools/org.lamport.tlatools/test/util/SimpleFilenameToStreamTest.java

```diff
@@ -186,6 +182,7 @@ public void testTLALibrarySystemProperty() throws IOException  {
 			System.setProperty(SimpleFilenameToStream.TLA_LIBRARY, d1.toString());
 			File f1 = resolver.resolve("Integers", true);
 			assertTrue("Resolver should still find Integers when library path is empty", f1.exists());
+			assertTrue("Integers is a standard module, even with TLA_LIBRARY set", resolver.isStandardModule("Integers"));
 
 			Files.writeString(d1.resolve("MyModule.tla"), "", StandardCharsets.US_ASCII);
 
```

Comment:
Some additional test cases based off weird behavior I've seen in SANY and TLC:
- The example of #545
- I've noticed that the current resolver won't find modules inside .jar files unless the .jar files go first in the classpath (which functions as SANY's search path currently); is that still the case or do these changes incidentally fix that?

Reply:
My goal is not to change any of TLC's weird module resolution rules, just to fix misuses of the Java resource API and do some cleanup. I might add a test case ensuring that this change _preserves_ any weird behavior from #545. 😆 

For your second point: maybe. See if #977 (merged in June last year) fixes the problem. If it doesn't, then this PR might incidentally.

Reply:
It seems that #545 has been fixed. I've added a regression test to this PR so we do not break that functionality in the future.

Reply:
> * I've noticed that the current resolver won't find modules inside .jar files unless the .jar files go first in the classpath (which functions as SANY's search path currently); is that still the case or do these changes incidentally fix that?

This is the intended behavior because it would be quite unusual to load the module override from the jar file but not the corresponding specification.